src/screens/Wallet.js

```diff
@@ -19,522 +19,505 @@ import WalletAddress from '../components/WalletAddress';
-  return {
-    selectedToken: state.selectedToken,
-    tokensHistory: state.tokensHistory,
-    tokensBalance: state.tokensBalance,
-    tokenMetadata: state.tokenMetadata || {},
-    tokens: state.tokens,
-    wallet: state.wallet,
-    walletState: state.walletState,
-    useWalletService: state.useWalletService,
-  };
-};
+import { useHistory } from "react-router-dom";
 
 
 /**
  * Main screen of the wallet
  *
  * @memberof Screens
  */
-class Wallet extends React.Component {
-  static contextType = GlobalModalContext;
-
-  constructor(props) {
-    super(props);
-
-    this.alertSuccessRef = React.createRef();
-  }
-
-  /**
-   * backupDone {boolean} if words backup was already done
-   * successMessage {string} Message to be shown on alert success
-   * shouldShowAdministrativeTab {boolean} If we should display the Administrative Tools tab
-   */
-  state = {
-    backupDone: true,
-    successMessage: '',
-    hasTokenSignature: false,
-    shouldShowAdministrativeTab: false,
-    totalSupply: null,
-    canMint: false,
-    canMelt: false,
-    transactionsCount: null,
-    mintCount: null,
-    meltCount: null,
-  };
-
-  // Reference for the unregister confirm modal
-  unregisterModalRef = React.createRef();
-
-  componentDidMount = async () => {
-    this.setState({
-      backupDone: LOCAL_STORE.isBackupDone()
-    });
-
-    this.initializeWalletScreen();
-  }
-
-  componentDidUpdate(prevProps) {
-    // the selected token changed, we should re-initialize the screen
-    if (this.props.selectedToken !== prevProps.selectedToken) {
-      this.initializeWalletScreen();
-    }
-
-    // if the new selected token history changed, we should fetch the token details again
-    const prevTokenHistory = get(prevProps.tokensHistory, this.props.selectedToken, {
-      status: TOKEN_DOWNLOAD_STATUS.LOADING,
-      updatedAt: -1,
-      data: [],
-    });
-    const currentTokenHistory = get(this.props.tokensHistory, this.props.selectedToken, {
+function Wallet() {
+  // Modal context
+  const context = useContext(GlobalModalContext);
+
+  // State
+  /** backupDone {boolean} if words backup was already done */
+  const [backupDone, setBackupDone] = useState(LOCAL_STORE.isBackupDone());
+  /** successMessage {string} Message to be shown on alert success */
+  const [successMessage, setSuccessMessage] = useState('');
+  /* shouldShowAdministrativeTab {boolean} If we should display the Administrative Tools tab */
+  const [shouldShowAdministrativeTab, setShouldShowAdministrativeTab] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(''); // TODO: Metadata token error are being suppressed as of now
+  const [totalSupply, setTotalSupply] = useState(null);
+  const [canMint, setCanMint] = useState(false);
+  const [canMelt, setCanMelt] = useState(false);
+  const [transactionsCount, setTransactionsCount] = useState(null);
+  const [mintCount, setMintCount] = useState(null);
+  const [meltCount, setMeltCount] = useState(null);
```

Comment:
[suggestion] Why not aggregate all the states in an object and use a single `useState`?

Reply:
I initially implemented it like that because it's the most common pattern for functional components: a getter/setter for each property.

Upon further investigation, I've seen that by having a state for each property we allow React to optimize when to re-render each element in the screen. If we had a single state object, whenever we updated any of its properties the whole screen would be re-rendered.

In practice, this seems to have almost no performance impact, so it's left as a style decision. I even thought about having `walletInfo: { mintCount, meltCount }` and `tokenInfo: { canMint, canMelt, txCount }` , but it didn't feel like it would improve the legibility of the code.

What do you think?

Reply:
I don't have much to say. If it is better to use this way let keep it.

Reply:
It's usually preferable to use each property as a different state so we can pass them as dependencies on `useCallback` and `useEffect` and other hooks, allowing react to memoize those hooks